ios/Locations.swift

```diff
@@ -1,145 +1,166 @@
+    }
+
+    asyncGroup.notify(queue: .main) {
+      print("Finished the loop - find \(geoList.count) addresses")
+      let results = self.aggregateAddressList(addressTSList: placeList)
+      callback([results])
+    }
   }
-  
 
   //---------------------------------------------------------------------------------
-  // Step2: aggregate address result - from timestamp to period
+  // MARK: - Step2: aggregate address result - from timestamp to period
   //---------------------------------------------------------------------------------
-  func aggregateAddressList(addressTSList:[AddressTS]) -> [String] {
-      var AddressPeriodList: [AddressPeriod] = []  //List of AddressPeriod
-      var start: Double = 0 // UTC start
-      var end: Double = 0 // UTC end
-      var address = "" // address pointer
-      for (index, address_ts) in addressTSList.enumerated(){
-          if index == 0 { // first one
-              start = address_ts.timestamp
-              end = address_ts.timestamp
-              address = address_ts.address
-          }
-          else if address_ts.address != address {
-              //append - dynamic size
-              //logic: same location + 10min
-              if start == end {
-                  AddressPeriodList.append(AddressPeriod(address:address,period: UTC_Converter(unixtime1:start,unixtime2:(start+600))))
-              } else {
-                  AddressPeriodList.append(AddressPeriod(address:address,period: UTC_Converter(unixtime1:start,unixtime2:end)))
-              }
-              //update
-              start = address_ts.timestamp
-              end = address_ts.timestamp
-              address = address_ts.address
-              if index == (addressTSList.count-1){ // last one - unique, add extra
-                  //logic: same location + 10min
-                  AddressPeriodList.append(AddressPeriod(address:address_ts.address, period: UTC_Converter(unixtime1:start,unixtime2:(start+600))))
-              }
-          } else {//move on
-              end = address_ts.timestamp
-              if index == (addressTSList.count-1){ // last one - same, add extra
-                  //logic: same location + 10min
-                  if start == end {
-                      AddressPeriodList.append(AddressPeriod(address:address, period: UTC_Converter(unixtime1:start,unixtime2:(start+600))))
-                  } else {
-                      AddressPeriodList.append(AddressPeriod(address:address, period: UTC_Converter(unixtime1:start,unixtime2:end)))
-                  }
-              }
+  func aggregateAddressList(addressTSList: [AddressTS]) -> [[String]] {
+    guard let first = addressTSList.first else { return [] }
+
+    var AddressPeriodList: [AddressPeriod] = []  // List of AddressPeriod
+    var start: Double = first.timestamp // UTC start
+    var end: Double = first.timestamp // UTC end
+    var previous = first // address pointer
+
+    let TEN_MINUTES_IN_MS: Double = 6 * 100 * 1000
```

Comment:
Should we randomize this "10 minutes" a little bit? (like between 9 to 11 minutes)?